File: SubscriptionApi/SubscriptionApi/Commands/AdapterFromSubscriptionCommandToCommand.cs
using SubscriptionApi.Dto;

namespace SubscriptionApi.Commands
{
  public interface CommandProcessing
  {
    void ApplyTo(SubscriptionCommand subscriptionStartCommand);
  }

  public class AdapterFromSubscriptionCommandToCommand : Command
  {
    private readonly SubscriptionCommand _innerCommand;

    public AdapterFromSubscriptionCommandToCommand(SubscriptionCommand innerCommand)
    {
      _innerCommand = innerCommand;
    }

    public void Invoke()
    {
      _innerCommand.ValidateData();
      _innerCommand.Resolve();
      _innerCommand.Authorize();
      _innerCommand.Invoke();
    }
  }
}

File: Java/factories/_01_encapsulation_of_global_context/src/main/java/com/github/TrainingExamples/grzesiek_galezowski/commands/AdapterFromSubscriptionCommandToCommand.java
package com.github.TrainingExamples.grzesiek_galezowski.commands;


public class AdapterFromSubscriptionCommandToCommand implements Command {
    private final SubscriptionCommand innerCommand;

    public AdapterFromSubscriptionCommandToCommand(SubscriptionCommand innerCommand) {
        this.innerCommand = innerCommand;
    }

    public void invoke() {
        innerCommand.validateData();
        innerCommand.resolve();
        innerCommand.authorize();
        innerCommand.invoke();
    }
}
